File: app/src/main/java/com/appsbygu/kadai_weather_java/adapters/PinpointLocationAdapter.java
package com.appsbygu.kadai_weather_java.adapters;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.appsbygu.kadai_weather_java.R;
import com.appsbygu.kadai_weather_java.models.apis.Weathers.PinpointLocation;

import io.realm.RealmList;

public class PinpointLocationAdapter extends RecyclerView.Adapter<PinpointLocationAdapter.PinpointViewHolder>{

    RealmList<PinpointLocation> pinpoints;

    public PinpointLocationAdapter(RealmList<PinpointLocation> pinpoints) {
        this.pinpoints = pinpoints;
    }

    @Override
    public PinpointLocationAdapter.PinpointViewHolder onCreateViewHolder(ViewGroup parent, int viewType){
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.pinpoint_location, parent, false);
        return new PinpointLocationAdapter.PinpointViewHolder(view);
    }

    @Override
    public void onBindViewHolder(PinpointLocationAdapter.PinpointViewHolder holder, int position){
        holder.ppNameTv.setText(pinpoints.get(position).getName());
        holder.ppLinkTv.setText(pinpoints.get(position).getLink());
    }

    @Override
    public int getItemCount() {
        return pinpoints.size();
    }

    class PinpointViewHolder extends RecyclerView.ViewHolder {
        private TextView ppNameTv;
        private TextView ppLinkTv;

        public PinpointViewHolder(View itemView){
            super(itemView);
            ppNameTv = (TextView) itemView.findViewById(R.id.ppNameTv);
            ppLinkTv = (TextView) itemView.findViewById(R.id.ppLinkTv);
        }
    }

}

File: app/src/main/java/com/appsbygu/kadai_weather_java/adapters/PinpointLocationAdapter.kt
package com.appsbygu.kadai_weather_java.adapters

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView

import com.appsbygu.kadai_weather_java.R
import com.appsbygu.kadai_weather_java.models.apis.weathers.PinpointLocation

import io.realm.RealmList

class PinpointLocationAdapter(var pinpoints: RealmList<PinpointLocation>) : RecyclerView.Adapter<PinpointLocationAdapter.PinpointViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): PinpointLocationAdapter.PinpointViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.pinpoint_location, parent, false)
        return PinpointLocationAdapter.PinpointViewHolder(view)
    }

    override fun onBindViewHolder(holder: PinpointLocationAdapter.PinpointViewHolder, position: Int) {
        holder.pinpointNameTextView.text = pinpoints[position]?.name
        holder.pinpointLinkTextView.text = pinpoints[position]?.link
    }

    override fun getItemCount(): Int {
        return pinpoints.size
    }

    class PinpointViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

        val pinpointNameTextView: TextView = itemView.findViewById(R.id.pinpointNameTextView)
        val pinpointLinkTextView: TextView = itemView.findViewById(R.id.pinpointLinkTextView)

    }

}
